akka-cluster/src/multi-jvm/scala/akka/cluster/MultiTeamClusterSpec.scala

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.cluster
+
+import akka.remote.testkit.{ MultiNodeConfig, MultiNodeSpec }
+import akka.remote.transport.ThrottlerTransportAdapter.Direction
+import com.typesafe.config.ConfigFactory
+
+import scala.concurrent.duration._
+
+object MultiTeamMultiJvmSpec extends MultiNodeConfig {
+  val first = role("first")
+  val second = role("second")
+  val third = role("third")
+  val fourth = role("fourth")
+  val fifth = role("fifth")
+
+  commonConfig(MultiNodeClusterSpec.clusterConfig)
+
+  nodeConfig(first, second)(ConfigFactory.parseString(
+    """
+      akka.cluster.team = "dc1"
+    """))
+
+  nodeConfig(third, fourth, fifth)(ConfigFactory.parseString(
+    """
+      akka.cluster.team = "dc2"
+    """))
+
+  testTransport(on = true)
+}
+
+class MultiTeamMultiJvmNode1 extends MultiTeamSpec
+class MultiTeamMultiJvmNode2 extends MultiTeamSpec
+class MultiTeamMultiJvmNode3 extends MultiTeamSpec
+class MultiTeamMultiJvmNode4 extends MultiTeamSpec
+class MultiTeamMultiJvmNode5 extends MultiTeamSpec
+
+abstract class MultiTeamSpec
+  extends MultiNodeSpec(MultiTeamMultiJvmSpec)
+  with MultiNodeClusterSpec {
+
+  import MultiTeamMultiJvmSpec._
+
+  "A cluster with multiple cluster teams" must {
+    "be able to form" in {
+
+      runOn(first) {
+        cluster.join(first)
+      }
+      runOn(second, third, fourth) {
+        cluster.join(first)
+      }
+      enterBarrier("form-cluster-join-attempt")
+
+      runOn(first, second, third, fourth) {
+        within(15.seconds) {
+          awaitAssert(clusterView.members.count(_.status == MemberStatus.Up) should ===(4))
+        }
+      }
+
+      enterBarrier("cluster started")
+    }
+
+    "have a leader per team" in {
+      runOn(first, second) {
+        cluster.settings.Team should ===("dc1")
+        clusterView.leader shouldBe defined
+        val dc1 = Set(address(first), address(second))
+        dc1.contains(clusterView.leader.get) should ===(true)
+      }
+      runOn(third, fourth) {
+        cluster.settings.Team should ===("dc2")
+        clusterView.leader shouldBe defined
+        val dc2 = Set(address(third), address(fourth))
+        dc2.contains(clusterView.leader.get) should ===(true)
+      }
+
+      enterBarrier("leader per team")
+    }
+
+    "be able to have team member changes while there is inter-team unreachability" in within(10.seconds) {
+      runOn(first) {
+        testConductor.blackhole(first, third, Direction.Both)
+      }
+      runOn(first, second, third, fourth) {
+        within(remaining) {
+          awaitAssert(clusterView.unreachableMembers.nonEmpty)
```

Comment:
it must be an assert (throwing something)
```
awaitAssert(clusterView.unreachableMembers should not be (Set.empty)
```

(always good to compare with collections instead of booleans because better error reporting)